zipline/sources/test_source.py

```diff
@@ -73,15 +73,19 @@ def advance_current(cur):
         """
         cur = cur + delta
 
-        if not (trading_schedule.is_executing_on_day
-                if daily_delta
-                else trading_schedule.is_executing_on_minute)(cur):
+        currently_executing = \
+            (daily_delta and (cur in trading_calendar.all_sessions)) or \
+            (trading_calendar.is_open_on_minute(cur))
+
+        if currently_executing:
+            return cur
+        else:
             if daily_delta:
-                return trading_schedule.next_execution_day(cur)
+                return trading_calendar.minute_to_session_label(cur)
             else:
-                return trading_schedule.next_start_and_end(cur)[0]
-        else:
-            return cur
+                return trading_calendar.open_and_close_for_session(
+                    trading_calendar.minute_to_session_label(cur)
+                )[0]
```

Comment:
I notice all these call to get the open and close just to get one.  Do you think we should add a `get_open` and `get_close`?